include/aktualizr-lite/aklite_client_ext.h

```diff
@@ -0,0 +1,64 @@
+#ifndef AKTUALIZR_LITE_CLIENT_EXT_H_
+#define AKTUALIZR_LITE_CLIENT_EXT_H_
+
+#include <cstdlib>
+#include <string>
+
+#include "json/json.h"
+
+#include "aktualizr-lite/tuf/tuf.h"
+
+#include "aktualizr-lite/api.h"
+
+class GetTargetToInstallResult {
+ public:
+  //
+  enum class Status {
+    // First block must match CheckInResult::Status
+    Ok = 0,    // check-in was good
+    OkCached,  // check-in failed, but locally cached meta-data is still valid
+    Failed,    // check-in failed and there's no valid local meta-data
+    NoMatchingTargets,
+    NoTargetContent,
+    SecurityError,
+    ExpiredMetadata,
+    MetadataFetchFailure,
+    MetadataNotFound,
+    BundleMetadataError,
+
+    // Additional values, specific for GetTargetToInstallResult
+    TufTargetNotFound = 100,
+  };
+
+  explicit GetTargetToInstallResult(const CheckInResult &check_in_result)
+      : status(static_cast<Status>(check_in_result.status)) {}
+
+  GetTargetToInstallResult(Status status, TufTarget selected_target, std::string reason)
+      : status(status), selected_target(std::move(selected_target)), reason(std::move(reason)) {}
+  Status status;
```

Comment:
Just idea. Maybe having two status fields in the `GetTargetToInstallResult` context would work better? Something like:


```
CheckInResult::Status checkin_status;
Status                find_target_status;
```